processing/src/main/java/io/druid/query/extraction/StringFormatExtractionFn.java

```diff
@@ -32,15 +33,47 @@
  */
 public class StringFormatExtractionFn extends DimExtractionFn
 {
+  public static enum NullHandling
+  {
+    NULLSTRING,
+    EMPTYSTRING,
+    RETURNNULL;
+
+    @JsonCreator
+    public static NullHandling forValue(String value)
+    {
+      return value == null ? NULLSTRING : NullHandling.valueOf(value.toUpperCase());
+    }
+
+    @JsonValue
+    public String toValue()
+    {
+      return name().toLowerCase();
+    }
+  }
+
   private final String format;
+  private final NullHandling nullHandling;
 
   @JsonCreator
   public StringFormatExtractionFn(
-      @JsonProperty("format") String format
+      @JsonProperty("format") String format,
+      @JsonProperty("nullHandling") NullHandling nullHandling
   )
   {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(format), "format string should not be empty");
     this.format = format;
+    this.nullHandling = nullHandling == null ? NullHandling.NULLSTRING : nullHandling;
+  }
+
+  public StringFormatExtractionFn(String format, String nullHandling)
```

Comment:
Are these extra constructors necessary? Seems like they're only used within the included test file, the forValue() and toValue() functions are enough for serde to work